internal/compose/compose.go

```diff
@@ -275,54 +275,58 @@ func (p *Project) WaitForHealthy(opts CommandOptions) error {
 		return err
 	}
 
+	timeout := time.After(10 * time.Minute)
+	ticker := time.NewTicker(1 * time.Second)
+
 	containerIDs := strings.Split(strings.TrimSpace(b.String()), "\n")
 	for {
 		if signal.SIGINT() {
 			return errors.New("SIGINT: cancel waiting for policy assigned")
 		}
 
-		healthy := true
-
-		logger.Debugf("Wait for healthy containers: %s", strings.Join(containerIDs, ","))
-		descriptions, err := docker.InspectContainers(containerIDs...)
-		if err != nil {
-			return err
-		}
+		select {
+		case <-timeout:
+			return errors.New("time out waiting for healthy container")
+		case <-ticker.C:
+			healthy := true
```

Comment:
Nit. Do nothing and continue in this `case`, to have the main part of the loop in the upper level.
```suggestion
		case <-ticker.C:
		}
		healthy := true
```